### PR TITLE
VM SCSI and SATA Adapter Samples

### PR DESCRIPTION
**Pull request by jobingeo**
Two samples VM SCSI  Adapter and SATA Adapter samples are added.
Here to pretty print/log the hash, I used .to_yaml method, which looks more neater than 'pp'
To run the samples
1.   .\run_sample.bat sata -l <VCIP> -u Administrator@vsphere.local -p Admin!23 -k --insecure -X --vm-name <Existing_VM_Name>
2.   .\run_sample.bat scsi -l <VCIP> -u Administrator@vsphere.local -p Admin!23 -k --insecure -X --vm-name <Existing_VM_Name>

### CODE REVIEW

File: client/samples/bin/launcher.rb [hunk 1]
```diff
@@ -8,6 +8,8 @@
     'create_basic_vm' => 'lib/sample/vcenter/vm/create/create_basic_vm.rb',
     'list_vms' => 'lib/sample/vcenter/vm/list_vms.rb',
     'power' => 'lib/sample/vcenter/vm/power.rb',
+	'sata' => 'lib/sample/vcenter/vm/hardware/adapter/sata.rb',
```
Comment (jrgarcia) [inline client/samples/bin/launcher.rb]: This line should be indented the same level as the rest.

File: client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  TITLE = 'SATA Configurations of a VM'
+  DESCRIPTION = <<EOL
+  Demonstrates how to configure virtual SATA adapters of a virtual machine.
+
+  Sample Prerequisites:
+  The sample needs an existing VM.
+EOL
+
+  # sample attributes
+  attr_reader :service_manager,:vm_name,:vm_id,:sata_svc,:satas_to_delete
```
Comment (jrgarcia) [inline client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb]: There should be a space between each comma and the next symbol.

File: client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+  def initialize
+    super(TITLE, DESCRIPTION, true)
+  end
+
+  def inject_options
+    option_parser.on("-v", "--vm-name NAME","Name of the VM") do |value|
+      options[:vm_name] =  value
+      end
```
Comment (jrgarcia) [inline client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb]: This should be indented the same level as line 27 (`option_parser.on("-v", ...`).

File: client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    ssl_options = {}
+    ssl_options[:verify] = :none if insecure
+    ssl_options[:ca_file] = ca_file if ca_file
+    @service_manager = get_service_manager(ssl_options)
+    @sata_svc = HARDWARE_ADAPTER_SATA_CLASS.new(service_manager.vapi_config)
+  end
+
+  def execute
+    @vm_id = VMHelper.get_vm(service_manager.vapi_config, @vm_name)
+    if not vm_id
```
Comment (jrgarcia) [inline client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb]: In most cases when you see `if not` or `if !`, you should use `unless` instead. In this case, this line would be `unless vm_id`.

File: client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      raise "Sample requires an existing vm with name #{vm_name}. '
+                        'Please create the vm first."
+    end
+    #Get Information about the Existing/Original SATA adapters of the VM
+    log.info "Using VM #{vm_name} for the SATA Configuration"
+    sata_summaries = sata_svc.list(vm_id)
+    (sata_summaries).each do|sata_summary|
```
Comment (jrgarcia) [inline client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb]: There should be a space after `do`.

File: client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+      sata = sata_summary.adapter
+      sata_info = sata_svc.get(vm_id, sata)
+      log.info"Original SATA Adapter Information for the vm #{vm_name} with the sata-id #{sata} is #{sata_info.to_yaml}"
```
Comment (jrgarcia) [inline client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb]: There should be a space after `log.info`.

File: client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    end
+
+    #Create a SATA adapter with the Default Configuration
+    log.info "# Example: Create SATA adapter with default configuration"
+    sata_create_spec = HARDWARE_ADAPTER_SATA_CLASS::CreateSpec.new()
+    sata = sata_svc.create(vm_id, sata_create_spec)
+    log.info("The sata controller created for the vm #{vm_name} is Sata #{sata}")
```
Comment (jrgarcia) [inline client/samples/lib/sample/vcenter/vm/hardware/adapter/sata.rb]: I'm fine with using parentheses or leaving them off, but it's nice to keep them consistent (`log.info` above has no parentheses).

**Comment by jrgarcia** [comment]
Thanks for these samples Jobin! This is awesome! Keep up the great work! :tada:

**Comment by jobingeo** [comment]
@jrgarcia   All requested changes requested are incorporated in sata.rb. The similar indendation/spacing issues were in scsi.rb. That were fixed too

**Pull request merged**